stream: implement fetch body mixin on Readable (#39520)
nodejs/node: doc/api/stream.md

The `'resume'` event is emitted when [`stream.resume()`][stream-resume] is
called and `readableFlowing` is not `true`.

##### `readable.arrayBuffer()`
<!--- YAML>
added: REPLACEME
--->

> Stability: 1 - Experimental

* Returns: {Promise}

> Returns a promise resolving to the full content as an array buffer.

Reviewer: ```suggestion
Returns a promise fulfilled with the full content as an `ArrayBuffer`.
```

##### `readable.blob()`
<!--- YAML>
added: REPLACEME
--->

> Stability: 1 - Experimental

* Returns: {Promise}

> Returns a promise resolving to the full content read as
> Blob.

Reviewer: ```suggestion
Returns a promise fulfilled with the full content as a {Blob}.
```

readable.isPaused(); // === false
```

##### `readable.json()`
<!--- YAML>
added: REPLACEME
--->

> Stability: 1 - Experimental

* Returns: {Promise}

> Returns a promise resolving to the full content read as
> utf8 string and parsed to JSON object.

Reviewer: ```suggestion
Returns a promise fulfulled with the full content read as an UTF-8
string and parsed as a JSON object.
```

});
```

##### `readable.text()`
<!--- YAML>
added: REPLACEME
--->

> Stability: 1 - Experimental

* Returns: {Promise}

> Returns a promise resolving to the full content read as
> utf8 string.

Reviewer: ```suggestion
Returns a promise fulfilled with the full content read as an UTF-8 string.
```
